app/src/main/java/com/example/parkingsystemjava/entity/Reservation.java

```diff
@@ -1,31 +1,65 @@
 package com.example.parkingsystemjava.entity;
 
-public class Reservation {
+import com.example.parkingsystemjava.utils.Constants;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
 
-    private static final String EMPTY_STRING = "";
+public class Reservation {
 
-    private String entryDate;
-    private String exitDate;
+    private Calendar entryDate;
+    private Calendar exitDate;
     private String keyCode;
 
     public Reservation() {
-        this.entryDate = EMPTY_STRING;
-        this.exitDate = EMPTY_STRING;
+        this.entryDate = null;
+        this.exitDate = null;
+    }
+
+    public Reservation(String entryDate, String exitDate, String keyCode){
+        this.entryDate = convertToCalendar(entryDate);
+        this.exitDate = convertToCalendar(exitDate);
+        this.keyCode = keyCode;
     }
 
     public void setEntryDate(String entryDate) {
-        this.entryDate = entryDate;
+        this.entryDate = convertToCalendar(entryDate);
     }
 
     public void setExitDate(String exitDate) {
-        this.exitDate = exitDate;
+        this.exitDate = convertToCalendar(exitDate);
     }
 
     public void setKeyCode(String keyCode) {
         this.keyCode = keyCode;
     }
 
-    public String getEntryDate() {
+    public Calendar getEntryDate() {
         return entryDate;
     }
+
+    public Calendar getExitDate() {
+        return exitDate;
+    }
+
+    public String getKeyCode() {
+        return keyCode;
+    }
+
+    private Calendar convertToCalendar(String dateString) {
```

Comment:
this method should be in the new DateUtils class

Reply:
done